core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -315,11 +321,23 @@ object ConsoleConsumer extends Logging {
 
     formatter.configure(formatterArgs.asScala.asJava)
 
-    val topicOrFilterOpt = List(topicOpt, whitelistOpt).filter(options.has)
-    if (topicOrFilterOpt.size != 1)
-      CommandLineUtils.printUsageAndDie(parser, "Exactly one of whitelist/topic is required.")
+    val filterOpts = List(includeOpt, whitelistOpt).filter(options.has)
+    if (filterOpts.size == 2) {
+      CommandLineUtils.warn("--whitelist is DEPRECATED and ignored if --include specified.")
```

Comment:
Checking this is a very good idea. However, I would `printUsageAndDie` when both are used together instead of ignoring `--whitelist`. We could say `Options --include and --whitelist cannot be specified together. --whitelist is DEPRECATED, use --include instead.`.

Reply:
I think with some other deprecated arguments we simply ignore the deprecated if the new one specified. for example

    --broker-list and --bootstrap-server in ConsoleProducer, VerifiableConsumer, VerifiableProducer and GetOffsetShell
    And the scripts don't fail if both deprecated and new one existed.

Reply:
I know. I don't think that we have been consistent about this but this does not prevent us from doing better here :). I am not a fan of having this warning printed out all the time for existing users. It might even break existing usages of the tool if someone does not expect it this new line.

Reply:
I think if someone started to use `--include` they already noticed the usage help message. But I got what you are saying. The other point is we already merged MirrorMaker script with only ignoring in this pr #10937 do we need to raise another one to handle this in the same way for consistency (keeping in mind that  KIP-629 suppose to be part of 3.0.0 release and the code freeze is tomorrow July 14th, 2021)

Reply:
The other option is to have another Jira ticket to change the ignoring for all deprecated arguments in case of the existence of the new ones to be handled probably later for another release.

Reply:
agree with staying consistent with other tools ignoring deprecated args if that's what we do elsewhere